workshop/exercise-7/README.md

```diff
@@ -0,0 +1,193 @@
+## (OPTIONAL, FOR ADVANCED USERS) Build and Deploy our Knative Application
+
+We've seen how to deploy an application from a container that already exists, so let's build the container image ourselves from source code. We'll be using Tekton Pipelines, a project providing Kubernetes-style resources for declaring CI/CD pipelines. We'll fetch our source code from a github repository, build a docker image from it, push it to a private container registry on IBM Cloud, and then ultimately get a URL to access our application.
+
+### Tekton Components Overview
+
+First, let's understand a few of the Tekton components we will be using in this exercise. A `Task` defines work that needs to be executed, and is made up of a series of `steps` that will be executed sequentially.
+
+A `TaskRun` will run the `Task` that you defined. You can then view output from the `TaskRun`.
+
+The main goal of Tekton Pipelines is to run your `Task` individually or as a part of a `Pipeline`. Each `Task` runs as a Pod on your Kubernetes cluster with each step as its own container.
+
+A `PipelineResource` is used to define artifacts that are passed into and out of a `Task`. There are a few sytem defined resource types ready to use, such as `git` or `image` resources.  
+
+### Create the Required PipelineResources
+
+1. Take a look at the `git-resource.yaml` file. 
+    ```
+    apiVersion: tekton.dev/v1alpha1
+    kind: PipelineResource
+    metadata:
+      name: fib-knative
+    spec:
+      type: git
+      params:
+        - name: revision
+          value: master
+        - name: url
+          value: https://github.com/IBM/fib-knative
+    ```
+    You can see that this file is defining the git resource we will be building. It includes the git url and revision for our code.
+
+2. Apply this yaml file to your kubernetes cluster.
+    ```
+    kubectl apply -f git-resource.yaml
+    ```
+
+3. Take a look at the `image-resource.yaml` file.
+    ```
+    apiVersion: tekton.dev/v1alpha1
+    kind: PipelineResource
+    metadata:
+      name: fib-knative
+    spec:
+      type: image
+      params:
+        - name: url
+          value: us.icr.io/<NAMESPACE>/fib-knative
+    ```
+    This file defines the location where our built image will be stored. In our case, we'll be storing it on the IBM Cloud Container Registry. 
+
+4. Make sure you update the image value with your own container registry namespace, which you defined in exercise 6.
```

Comment:
for those who need hand-holding... mention they need to replace the `<NAMESPACE>` stuff.